src/libraries/System.DirectoryServices.AccountManagement/src/System/DirectoryServices/AccountManagement/AD/ADStoreCtx.cs

```diff
@@ -2418,12 +2421,22 @@ protected virtual void LoadDomainInfo()
             this.dnsHostName = ADUtils.GetServerName(this.ctxBase);
 
             // Pull the requested port number
-            Uri ldapUri = new Uri(this.ctxBase.Path);
-            int port = ldapUri.Port != -1 ? ldapUri.Port : (ldapUri.Scheme.ToUpperInvariant() == "LDAPS" ? 636 : 389);
+            int port = LdapDefaultPort;
+            if (Uri.TryCreate(ctxBase.Path, UriKind.Absolute, out Uri ldapUri))
+            {
+                if (ldapUri.Port != -1)
+                {
+                    port = ldapUri.Port;
+                }
+                else if (string.Equals(ldapUri.Scheme, "LDAPS", StringComparison.OrdinalIgnoreCase))
```

Comment:
Another issue from the original PR - it seems odd to me that we ignore this scheme and instead use LDAP below with a port.  Do you know if that's intentional?

Reply:
i.e. should we use "LDAPS://" below when `ldapUri.Scheme == "LDAPS"`? Would the connection bel secure \ encrypted just because it's on the LDAPS port even though "LDAP://" is used?

Reply:
It has been like this from day one. After digging a little deeper and talking with my team, I found out that LDAPS is not valid provider name supported. Hence it uses LDAP as the supported provider name. This allows the user to call the api with LDAPS and the api in turn uses the ssl port(636) along with the supported provider name which LDAP. This is a less strict way to call the api and allow ssl port. Other way would not allow LDAPS at all which might break a lot of applications.